signac/sync.py

```diff
@@ -254,68 +259,58 @@ def sync_jobs(src, dst, strategy=None, exclude=None, doc_sync=None, recursive=Fa
-            a :class:`.errors.SyncConflict` exception will be raised upon conflict.
-        :param exclude:
-            A filename exclusion pattern. All files matching this pattern will be
-            excluded from the synchronization process.
-        :type exclude:
-            str
-        :param doc_sync:
-            A synchronization strategy for document keys. The default is to use a safe key-by-key
-            strategy that will not overwrite any values on conflict, but instead raises a
-            :class:`~.errors.DocumentSyncConflict` exception.
-        :param recursive:
-            Recursively synchronize sub-directories encountered within
-            the job workspace directories.
-        :type recursive:
-            bool
-        :param follow_symlinks:
-            Follow and copy the target of symbolic links.
-        :type follow_symlinks:
-            bool
-        :param preserve_permissions:
-            Preserve file permissions
-        :type preserve_permissions:
-            bool
-        :param preserve_times:
-            Preserve file modification times
-        :type preserve_times:
-            bool
-        :param preserve_owner:
-            Preserve file owner
-        :type preserve_owner:
-            bool
-        :param preserve_group:
-            Preserve file group ownership
-        :type preserve_group:
-            bool
-        :param dry_run:
-            If True, do not actually perform any synchronization operations.
+    By default, this method will synchronize all files and document data
+    of dst job with the src job until a synchronization conflict occurs.
+    There are two different kinds of synchronization conflicts:
+
+        1. The two jobs have files with the same name, but different content.
+        2. The two jobs have documents that share keys, but those keys are
+           mapped to different values.
+
+    A file conflict can be resolved by providing a 'FileSync' *strategy* or by
+    *excluding* files from the synchronization. An unresolvable conflict is indicated
+    with the raise of a :class:`~.errors.FileSyncConflict` exception.
+
+    A document synchronization conflict can be resolved by providing a doc_sync function
+    that takes the source and the destination document as first and second argument.
+
+    Parameters
+    ----------
+    src : :class:`~signac.contrib.job.Job`
+        The src job, data will be copied from this job's workspace.
+    dst : :class:`~signac.contrib.job.Job`
+        The dst job, data will be copied to this job's workspace.
+    strategy :
+        A synchronization strategy for file conflicts. If no strategy is
+        provided, a :class:`.errors.SyncConflict` exception will be raised upon
+        conflict. (Default value = None)
+    exclude : str
+        A filename exclusion pattern. All files matching this pattern will be
+        excluded from the synchronization process. (Default value = None)
+    doc_sync :
```

Comment:
Do `doc_sync` and `strategy` need a type?

Reply:
I have no idea what to put for the type of `doc_sync`. It supports `None`, `False`, `'copy'`, and callables. Users are supposed to provide one of the properties/functors from the `DocSync` class, but I'm not sure how to express that as a type. @csadorf Do you have thoughts? Maybe this?
```suggestion
    doc_sync : attribute or callable provided by :py:class:`~signac.sync.DocSync`
```

Reply:
I went ahead and committed a slightly edited version of that suggestion above.

Reply:
I think that's fine, but maybe you can add an example on how to use this function as part of the general description? I think that would make it much easier for users to understand how to use this.